app_dart/test/request_handlers/get_green_commits_test.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:cocoon_service/src/model/appengine/commit.dart';
+import 'package:cocoon_service/src/model/appengine/stage.dart';
+import 'package:cocoon_service/src/model/appengine/task.dart';
+import 'package:cocoon_service/src/service/build_status_provider.dart';
+import 'package:cocoon_service/src/service/datastore.dart';
+import 'package:gcloud/db.dart';
+import 'package:test/test.dart';
+
+import '../src/datastore/fake_config.dart';
+import '../src/request_handling/fake_authentication.dart';
+import '../src/request_handling/request_handler_tester.dart';
+import '../src/service/fake_build_status_provider.dart';
+
+void main() {
+  group('GetStatus', () {
+    late FakeConfig config;
+    FakeClientContext clientContext;
+    FakeKeyHelper keyHelper;
+    FakeBuildStatusService buildStatusService;
+    late RequestHandlerTester tester;
+    late GetGreenCommits handler;
+
+    late Commit commit1;
+    late Commit commit2;
+
+    late Task task1;
+    late Task task2;
+    late Task task3;
+    late Task task4;
+    late Task task5;
+
+    late Stage stage1;
+    late Stage stage2;
+    late Stage stage3;
+    late Stage stage4;
+    late Stage stage5;
+
+    Future<T?> decodeHandlerBody<T>() async {
+      final Body body = await tester.get(handler);
+      return await utf8.decoder.bind(body.serialize() as Stream<List<int>>).transform(json.decoder).single as T?;
+    }
+
+    setUp(() {
+      clientContext = FakeClientContext();
+      keyHelper = FakeKeyHelper(applicationContext: clientContext.applicationContext);
+      tester = RequestHandlerTester();
+      config = FakeConfig(keyHelperValue: keyHelper);
+      buildStatusService = FakeBuildStatusService(commitStatuses: <CommitStatus>[]);
+      handler = GetGreenCommits(
+        config,
+        datastoreProvider: (DatastoreDB db) => DatastoreService(config.db, 5),
+        buildStatusProvider: (_) => buildStatusService,
+      );
+      commit1 = Commit(
```

Comment:
Take advantage of the `entity_generators.dart` class. It allows you to do `generateCommit` and `generateTask` to remove the boiler plate